lib/datadog/appsec/waf.rb

```diff
@@ -576,18 +581,6 @@ def valid!
 
           fail LibDDWAF::Error, "Attempt to use an invalid instance: #{inspect}"
         end
-
-        def retained
-          @retained ||= []
-        end
-
-        def retain(object)
-          retained << object
-        end
-
-        def release(object)
-          retained.delete(object)
-        end
```

Comment:
I've only removed the `retained` API from the `Waf::Handle` since is the only class I touched on this PR. 

I agree we could get rid of the `Waf::Context` as well. I will make a note an address it on a separate PR